Userland/Libraries/LibTest/Randomized/RandomRun.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2023, Martin Janiczek <martin@janiczek.cz>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/QuickSort.h>
+#include <AK/Vector.h>
+
+#include <LibTest/Randomized/Chunk.h>
+
+namespace Test {
+namespace Randomized {
+
+// RandomRun is a record of random bits used in generation of random values.
+// Once a value failing user test is found, we then attempt to shrink its
+// RandomRun using various ShrinkCommands.
+//
+// This means that we construct new RandomRuns by saying "OK, but what if the
+// PRNG gave you 0 instead of 23 that time..." The runner then tries to
+// generate a new value from the new RandomRun; if it succeeds and the value
+// still fails the test, we've shrunk our counterexample some!
+//
+// RandomRun is conceptually a sequence of unsigned integers, e.g.
+// [5,3,10,8,0,0,1].
+class RandomRun {
+public:
+    RandomRun() { m_data = Vector<u32>(); }
+    RandomRun(RandomRun const& rhs)
```

Comment:
I'm still not 100% sure we _don't_ need the resetting of the index (a new RandomRun, mainly after shrinking, should always be run from beginning, instead of from the middle), but ... from the user perspective everything still seems to work. I'm changing these to be default constructors instead, but ... again, not 100% sure. Maybe if I stepped through the codebase I'd find we should rather use a named operation that resets the "iterator" index.